cmd/kube-rbac-proxy/app/options/proxyoptions.go

```diff
@@ -125,13 +140,87 @@ func (o *ProxyOptions) ApplyTo(c *server.KubeRBACProxyInfo, a *serverconfig.Auth
 		}
 	}
 
+	c.UpstreamHeaders = o.UpstreamHeader
 	c.IgnorePaths = o.IgnorePaths
 	c.AllowPaths = o.AllowPaths
 	a.APIAudiences = o.TokenAudiences
 
 	return nil
 }
 
+func validateSecureConnection(o *ProxyOptions) error {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Duration(o.UpstreamDNSTimeout)*time.Second)
+	defer cancel()
+
+	isLoopback, err := isLoopbackAddress(ctx, o.Upstream)
+	if err == nil && isLoopback {
+		return nil
+	}
+
+	if o.UpstreamForceH2C {
+		return fmt.Errorf("loopback address is required for h2c")
```

Comment:
the above error is unhandled, we may still be on a loopback, so this error message might be false

Reply:
Good catch.

Reply:
Let me handle the error consciously as it might create the impression, that a mTLS connection is necessary for identity headers, even though a loopback address would be fine as well.

Reply:
> If it is not a loopback address or there is an error, h2c shall not work and we continue validating for mTLS.
> If the config enables mTLS, we don't need loopback for identity headers.

right, but even in those cases you should at least log the error for more clarity